File: Golang/DataObjects/CreateDataObject.go
package main

import (
    "bytes"
    "encoding/json"
    "fmt"
    "net/http"
    "time"
)

func main() {
    boardsOnFireApiUrl := "YOUR URL" // example: https://my-company.boardsonfireapp.com/api/v5/
    apiKey := "YOUR API KEY"
    dataSourceName := "YOUR DATA SOURCE NAME"
    organizationId := "YOUR ORGANIZATION ID"

    client := &http.Client{}

    // Data Source with numeric fields goal and outcome
    requestBody := map[string]interface{}{
        "goal":       50,
        "outcome":    40,
        "timestamp":  "2024-11-08T10:30:00Z",
        "group_name": "default",
        "comment":    "Sample comment",
        "data_annotations": map[string]interface{}{
            "outcome": map[string]interface{}{
                "comment": "Outcome comment",
                "color":   "blue",
            },
        },
        "organization_id": organizationId,
    }

    jsonContent, err := json.Marshal(requestBody)
    if err != nil {
        fmt.Println("Error marshalling request body:", err)
        return
    }

    req, err := http.NewRequest("POST", fmt.Sprintf("%s/datasources/%s/dataobjects", boardsOnFireApiUrl, dataSourceName), bytes.NewBuffer(jsonContent))
    if err != nil {
        fmt.Println("Error creating request:", err)
        return
    }

    req.Header.Set("Content-Type", "application/json")
    req.Header.Set("x-api-key", apiKey)

    resp, err := client.Do(req)
    if err != nil {
        fmt.Println("Request error:", err)
        return
    }
    defer resp.Body.Close()

    if resp.StatusCode != http.StatusOK {
        fmt.Println("Request failed with status:", resp.Status)
        return
    }

    var data map[string]interface{}
    if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
        fmt.Println("Error decoding response:", err)
        return
    }

    fmt.Println("DataObject:")
    fmt.Println(data)
}

File: Python/DataObjects/CreateDataObject.py
import requests
import json
from datetime import datetime

boards_on_fire_api_url = "YOUR URL"  # example: https://my-company.boardsonfireapp.com/api/v5/
api_key = "YOUR API KEY"
data_source_name = "YOUR DATA SOURCE NAME"
organization_id = "YOUR ORGANIZATION ID"

headers = {
    "Content-Type": "application/json",
    "x-api-key": api_key
}

request_body = {
    "goal": 50,
    "outcome": 40,
    "timestamp": "2024-11-08T10:30:00Z",
    "group_name": "default",
    "comment": "Sample comment",
    "data_annotations": {
        "outcome": {
            "comment": "Outcome comment",
            "color": "blue"
        }
    },
    "organization_id": organization_id
}

try:
    response = requests.post(f"{boards_on_fire_api_url}/datasources/{data_source_name}/dataobjects", headers=headers, data=json.dumps(request_body))
    response.raise_for_status()
    data = response.json()
    print("DataObject:")
    print(data)
except requests.exceptions.RequestException as e:
    print(f"Request error: {e}")
